### PR TITLE
Added tsebra

### LINKED ISSUE #5464: new module: TSEBRA
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by GallVp**
## PR checklist

Closes #5464 

- [x] Added tsebra
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/tsebra/main.nf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/tsebra:1.1.2.4--pyhca03a8a_0':
+        'biocontainers/tsebra:1.1.2.4--pyhca03a8a_0' }"
+
+    input:
+    tuple val(meta), path(gtfs)
+    path hints_files
```
Comment (SPPearce) [inline modules/nf-core/tsebra/main.nf:12]: Are these compulsory?
And do they need a meta?

File: modules/nf-core/tsebra/main.nf [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    path keep_gtfs
```
Comment (SPPearce) [inline modules/nf-core/tsebra/main.nf:13]: Is this a path or a value?

File: modules/nf-core/tsebra/main.nf [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args        = task.ext.args                                     ?: ''
+    def prefix      = task.ext.prefix                                   ?: "${meta.id}"
+    def gtf_arg     = '-g ' + gtfs.collect { "$it" }.join(',')
+    def hints_arg   = '-e ' + hints_files.collect { "$it" }.join(',')
+    def keep_arg    = keep_gtfs                                         ? ( '-k ' + keep_gtfs.collect { "$it" }.join(',') ) : ''
+    def config_arg  = config                                            ? "-c $config"                                      : ''
+    def VERSION     = '1.1.2.4'
```
Comment (SPPearce) [inline modules/nf-core/tsebra/main.nf:31]: Please add the standard disclaimer that the VERSION string needs updating if bumping the version.

File: modules/nf-core/tsebra/main.nf (hunk 2 above)
Comment (GallVp) [reply modules/nf-core/tsebra/main.nf:13]: `keep_gtfs` is a list of paths similar to `hints_files`.  It starts with a verb and can be confusing. But that's the keyword `keep_gtf` used by tesbra in its code and docs:

https://github.com/Gaius-Augustus/TSEBRA/blob/c87ba3a82df40953d2a9bc19f770b676ad389949/bin/tsebra.py#L173

File: modules/nf-core/tsebra/main.nf (hunk 3 above)
Comment (GallVp) [reply modules/nf-core/tsebra/main.nf:31]: Can you kindly help with review of the test data?

https://github.com/nf-core/test-datasets/pull/1176

File: modules/nf-core/tsebra/main.nf (hunk 2 above)
Comment (SPPearce) [reply modules/nf-core/tsebra/main.nf:13]: Hmm, still not very clear here though. 
So there is a gtf in the main input as well as this keep_gtfs?

Comment (GallVp) [reply modules/nf-core/tsebra/main.nf:13]: I'll try to elaborate further.

- `gtfs`: GTFs containing proposed annotations. If an annotation does not pass the evaluation process, it is dropped.
- `hints_files`: GFF files containing external evidence such as intron and cds hints to evaluate the proposed annotations.
- `keep_gtfs`:  GTFs containing established annotations. These annotations are included in the evaluation process but not excluded.

Comment (SPPearce) [reply modules/nf-core/tsebra/main.nf:14]: Ok, fine.

**Review by SPPearce — APPROVED** [review]

**Pull request merged**